GiniSwitchSDK/GiniSwitchSDK/Classes/ExtractionValue.swift

```diff
@@ -7,14 +7,21 @@
 //
 
 public struct ContractAddressValue : ExtractionValue {
-    public var name:String
-    public var city:String
-    public var postalCode:String
-    public var country:String
-    public var street:AddressValue
+    public var name:String?
+    public var city:String?
+    public var postalCode:String?
+    public var country:String?
+    public var street:AddressValue?
     
     public var valueString: String {
-        return "\(name), \(street.valueString), \(city), \(postalCode), \(country)"
+        let components = [name, street?.valueString, city, postalCode, country].flatMap({
+            if let component = $0,
+                !component.isEmpty {
+                return component
+            }
+            return nil
+        })
+        return components.joined(separator: ", ")
```

Comment:
This `flatMap` can be replaced by a `reduce`, and the `joined` would dissappear in that case. Maybe is more clear to have this in two steps, as you prefer.

Reply:
I thought about it, but I'd need an additional if statement to remove the last comma. That's what I wanted to avoid. The Joined method will be smart enough to remove it.